pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -235,25 +238,25 @@ func (f *ring0Factory) JSONEncoder() runtime.Encoder {
 	return legacyscheme.Codecs.LegacyCodec(legacyscheme.Registry.EnabledVersions()...)
 }
 
-func (f *ring0Factory) UpdatePodSpecForObject(obj runtime.Object, fn func(*api.PodSpec) error) (bool, error) {
+func (f *ring0Factory) UpdatePodSpecForObject(obj runtime.Object, fn func(*v1.PodSpec) error) (bool, error) {
 	// TODO: replace with a swagger schema based approach (identify pod template via schema introspection)
 	switch t := obj.(type) {
```

Comment:
The number of objects and the number of versions reveals the unsustainability of this approach (especially as the workload API objects are getting promoted to new API versions - @kow3ns). This is ok for now, but you should add a reflective unit test that will fail if a compiled-in versioned type containing a pod spec is added and not handled here. Long-term, we need to find a way for kubectl to discover and extract pod spec fields from arbitrary objects that are not compiled in.

Reply:
missing appsv1.Deployment, appsv1.StatefulSet, appsv1beta1.ReplicaSet, appsv1.ReplicaSet

Reply:
this is constantly going to drift as new versions of types are added

Reply:
thanks you ～ add it now. is it possible can have accessor return v1.podtemplate?
```
type ObjectMetaAccessor interface {
	GetObjectMeta() Object
}

// Object lets you work with object metadata from any of the versioned or
// internal API objects. Attempting to set or retrieve a field on an object that does
// not support that field (Name, UID, Namespace on lists) will be a no-op and return
// a default value.
type Object interface {
	GetNamespace() string
	SetNamespace(namespace string)
	GetName() string
	SetName(name string)
	GetGenerateName() string
	SetGenerateName(name string)
	GetUID() types.UID
	SetUID(uid types.UID)
	GetResourceVersion() string
	SetResourceVersion(version string)
	GetGeneration() int64
	SetGeneration(generation int64)
	GetSelfLink() string
	SetSelfLink(selfLink string)
	GetCreationTimestamp() Time
	SetCreationTimestamp(timestamp Time)
	GetDeletionTimestamp() *Time
	SetDeletionTimestamp(timestamp *Time)
	GetDeletionGracePeriodSeconds() *int64
	SetDeletionGracePeriodSeconds(*int64)
	GetLabels() map[string]string
	SetLabels(labels map[string]string)
	GetAnnotations() map[string]string
	SetAnnotations(annotations map[string]string)
	GetInitializers() *Initializers
	SetInitializers(initializers *Initializers)
	GetFinalizers() []string
	SetFinalizers(finalizers []string)
	GetOwnerReferences() []OwnerReference
	SetOwnerReferences([]OwnerReference)
	GetClusterName() string
	SetClusterName(clusterName string)
}
```

Reply:
> is it possible can have accessor return v1.podtemplate?

no, ObjectMetaAccessor only returns things that are part of metadata that all objects have